### DISCUSSION TITLE
Is `postscan` the same as `scan` but with the first element dropped?

### DISCUSSION THREAD

**Discussion by eddiemundo** [composewell/streamly#2731] [Q&A] [answered]
I'm not sure since the implementation of `scan` and `postscan` are different looking.

**Comment by shlok** [comment, marked as answer]
Yes (if I understand you correctly). `postscan` drops the extraction of the initial fold state, but `scan` doesn't. (In practice, I have only used `postscan` because that simply yields an output for each input, whereas `scan` yields this one extra thing in the beginning.)
